vsphere/resource_vsphere_datacenter_test.go

```diff
@@ -81,6 +81,35 @@ resource "vsphere_datacenter" "testDC" {
 }
 `
 
+const testAccCheckVSphereDatacenterConfigAttributes = `
+resource "vsphere_custom_attribute" "terraform-test-attribute" {
+  name                = "terraform-test-attribute"
+  managed_object_type = "Datacenter"
+}
+
+resource "vsphere_datacenter" "testDC" {
+  name = "testDC"
+  custom_attributes = "${map(vsphere_custom_attribute.terraform-test-attribute.id, "value")}"
```

Comment:
I notice the pattern of using the `map` interpolation function is present in tests quite a bit. I'm not too sure if this works, but have you just tried:

```hcl
resource "vsphere_datacenter" "testDC" {
  name = "testDC"
  custom_attributes = {
    "${vsphere_custom_attribute.terraform-test-attribute.id}" = "value"
  }
}
```

I would prefer that if it works - leaving more complicated scenarios using interpolation to any individual practitioner's use case :slightly_smiling_face:

Reply:
Yeah, use of the `map` function bothers me as well; but unfortunately interpolation doesn't work for map keys in resources as mentioned here https://github.com/hashicorp/terraform/issues/14516#issuecomment-301630345 :disappointed:.

One work around I often use is to place the definition of the custom attribute map in a locals block, where map keys are interpolated (keys will also be interpolated when passing to a module) as follows:
```hcl
locals {
  dc_attrs = {
    "${vsphere_custom_attribute.terraform-test-attribute.id}" = "value"
  }
}

resource "vsphere_datacenter" "testDC" {
  name = "testDC"
  custom_attributes = ${local.dc_attrs}
}
```
If you would prefer I can update the tests to work that way.

Reply:
I think this looks a lot cleaner :slightly_smiling_face: and we also use the map assignment pattern in other places like you type there (not necessarily in the vSphere provider, but elsewhere in TF-land).